pxtsim/runtime.ts

```diff
@@ -311,6 +311,13 @@ namespace pxsim {
             })
             return v;
         }
+
+        // Copied from pxtlib/util.ts
+        export function sanitizeFileName(name: string): string {
+            /* eslint-disable no-control-regex */
+            return name.replace(/[()\\\/.,?*^:<>!;'#$%^&|"@+=«»°{}\[\]¾½¼³²¦¬¤¢£~­¯¸`±\x00-\x1F]/g, '').trim().replace(/\s+/g, '-');
```

Comment:
Two questions for this regex. Is there not a way to capture all special characters without something like this? Also, how were you able to type some of these?

Reply:
This method was copy/pasted from pxtlib/util.ts, because that module is not available in the simulator. I don't know the history of this regex, but wanted to use it for consistency.

Reply:
I agree the regex could be written a lot better. It should take the approach of defining what is allowed rather than what is not.

Reply:
Your comment made me realize that css has slightly different validation rules than filenames. css names can't start with a number, for example. Pushed an update to fix this.